packages/teams-js/src/internal/videoPerformanceMonitor.ts

```diff
@@ -0,0 +1,95 @@
+import { inServerSideRenderingEnvironment } from '../private/inServerSideRenderingEnvironment';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { PerformanceStatistics } from './performanceStatistics';
+
+export class VideoPerformanceMonitor {
+  private static readonly distributionBinSize = 1000;
+  private static readonly calculateFPSInterval = 1000;
+
+  private isFirstFrameProcessed = false;
+  private currentSelectedEffect: {
+    effectId: string;
+    effectParam?: string;
+  };
+
+  private frameProcessTimeLimit = 100;
+  private startGettingTextureStreamTime: number;
+  private currentSteamId: string;
+  private frameProcessingStartedAt = 0;
+  private frameProcessingTimeCost = 0;
+  private processedFrameCount = 0;
+
+  private performanceStatistics: PerformanceStatistics;
+
+  public constructor(private reportPerformanceEvent: (actionName: string, args: unknown[]) => void) {
+    if (inServerSideRenderingEnvironment()) {
```

Comment:
Let's remove the error from constructor.

Reply:
removed